contentcuration/contentcuration/models.py

```diff
@@ -1153,40 +1183,51 @@ def _orphan_tree_id_subquery(cls):
     @classmethod
     def filter_edit_queryset(cls, queryset, user=None, user_id=None):
         user_id = user_id or not user.is_anonymous() and user.id
-        user_queryset = User.objects.filter(id=user_id)
 
-        queryset = queryset.annotate(
-            edit=Exists(user_queryset.filter(cls._edit_filter)),
-        )
+        queryset = queryset.exclude(pk=settings.ORPHANAGE_ROOT_ID)
+        orphan_filter = Q(tree_id=cls._orphan_tree_id_subquery())
+
+        if not user_id:
+            return queryset.annotate(edit=False).filter(orphan_filter)
 
-        queryset = queryset.filter(Q(edit=True) | Q(tree_id=cls._orphan_tree_id_subquery()))
+        edit_cte = PermissionCTE.editable_channels(user_id)
 
-        return queryset.exclude(pk=settings.ORPHANAGE_ROOT_ID)
+        return queryset.with_cte(edit_cte).annotate(
+            edit=edit_cte.exists(cls._permission_filter),
+        ).filter(Q(edit=True) | orphan_filter)
 
     @classmethod
     def filter_view_queryset(cls, queryset, user):
         user_id = not user.is_anonymous() and user.id
-        user_queryset = User.objects.filter(id=user_id)
 
         queryset = queryset.annotate(
-            edit=Exists(user_queryset.filter(cls._edit_filter)),
-            view=Exists(user_queryset.filter(cls._view_filter)),
             public=Exists(
                 Channel.objects.filter(
                     public=True, main_tree__tree_id=OuterRef("tree_id")
-                )
+                ).values("pk")
             ),
+        ).exclude(pk=settings.ORPHANAGE_ROOT_ID)
+
+        orphan_filter = Q(tree_id=cls._orphan_tree_id_subquery())
+
+        if not user_id:
+            return queryset.annotate(edit=False, view=False).filter(Q(public=True) | orphan_filter)
```

Comment:
Likewise here!

Can probably also skip anonymous access to the orphanage, and only return public nodes.

Reply:
Ah thanks, that makes sense! :sweat_smile: